library/core/src/intrinsics.rs

```diff
@@ -1579,175 +1667,157 @@ extern "rust-intrinsic" {
     #[rustc_nounwind]
     #[rustc_diagnostic_item = "intrinsics_unaligned_volatile_store"]
     pub fn unaligned_volatile_store<T>(dst: *mut T, val: T);
+}
 
+// Float operations, unsafe functions
+intrinsics! {
     /// Returns the square root of an `f32`
     ///
     /// The stabilized version of this intrinsic is
     /// [`f32::sqrt`](../../std/primitive.f32.html#method.sqrt)
-    #[rustc_nounwind]
-    pub fn sqrtf32(x: f32) -> f32;
+    // #[rustc_nounwind]
+    pub unsafe fn sqrtf32(_x: f32) -> f32;
     /// Returns the square root of an `f64`
     ///
     /// The stabilized version of this intrinsic is
     /// [`f64::sqrt`](../../std/primitive.f64.html#method.sqrt)
-    #[rustc_nounwind]
-    pub fn sqrtf64(x: f64) -> f64;
+    // #[rustc_nounwind]
+    pub unsafe fn sqrtf64(_x: f64) -> f64;
 
     /// Raises an `f16` to an integer power.
     ///
     /// The stabilized version of this intrinsic is
     /// [`f16::powi`](../../std/primitive.f16.html#method.powi)
-    #[rustc_nounwind]
-    pub fn powif16(a: f16, x: i32) -> f16;
+    pub unsafe fn powif16(_a: f16, _x: i32) -> f16;
```

Comment:
I think these float intrinsics could probably just be safe if there are no preconditions, not as part of this PR though.